tests/test_requests.py

```diff
@@ -1303,57 +1303,75 @@ def test_response_lines(self):
         might not be lined up ideally.
         """
         mock_chunks = [
-            'This \r\n',
-            '',
-            'is\r',
-            '\n',
-            'a',
-            ' ',
-            '',
-            '',
-            'test.',
-            '\r',
-            '\n',
-            'end.',
+            b'This \r\n',
+            b'',
+            b'is\r',
+            b'\n',
+            b'a',
+            b' ',
+            b'',
+            b'',
+            b'test.',
+            b'\r',
+            b'\n',
+            b'end.',
         ]
-        mock_data = ''.join(mock_chunks)
+        mock_data = b''.join(mock_chunks)
+        unicode_mock_data = mock_data.decode('utf-8')
 
-        mock_iter_content = lambda *args, **kwargs: (e for e in mock_chunks)
+        def mock_iter_content(*args, **kwargs):
+            if kwargs.get("decode_unicode"):
+                return (e.decode('utf-8') for e in mock_chunks)
+            return (e for e in mock_chunks)
 
         r = requests.Response()
         r._content_consumed = True
         r.iter_content = mock_iter_content
 
-        assert list(r.iter_lines(delimiter='\r\n')) == mock_data.split('\r\n')
+        # decode_unicode=None, output raw bytes
+        assert list(r.iter_lines(delimiter=b'\r\n')) == mock_data.split(b'\r\n')
 
-        # Because '\n' is a single line-end, when `iter_lines()` receives
-        # the chunks containing a single '\n', it emits '' as a line -- whereas
-        # `.splitlines()` combines with the '\r' and splits on `\r\n`.
+        # decode_unicode=True, output unicode strings
+        assert list(r.iter_lines(decode_unicode=True, delimiter=u'\r\n')) == unicode_mock_data.split(u'\r\n')
+
+        # When delimiter is None, we should yield the same result as splitlines()
+        # which supports the universal newline.
+        # '\r', '\n', and '\r\n' are all treated as one line break.
+
+        # decode_unicode=None, output raw bytes
         result = list(r.iter_lines())
-        assert result != mock_data.splitlines()
-        assert result[2] == ''
-        assert result[4] == ''
+        assert result == mock_data.splitlines()
```

Comment:
We can probably collapse this line into the one above it as with the other assertions:

```python
assert list(r.iter_lines()) == mock_data.splitlines()
```